MaxText/inference_utils.py

```diff
@@ -172,3 +174,49 @@ def sample_topk_logits(logits, topk, temperature, rng):
   topk_token = jnp.expand_dims(jax.random.categorical(rng, topk_logits / temperature).astype(jnp.int32), axis=-1)
   sampled_tokens = jnp.squeeze(jnp.take_along_axis(topk_idxs, topk_token, axis=-1), axis=-1).astype(jnp.int32)
   return sampled_tokens
+
+
+def sample_topk_topp_weighted(logits, topk, nucleus_topp, temperature, rng):
```

Comment:
My understanding is that while server is running, users could pass in one config of any choice, and do the sampling? If so, is that possible to recall existing func, like `sample_topk_logits`, `sample_nucleus_topp_logits`, etc using if/else? 

Or users could use all sampling methods in one single benchmark? I didn't see early return after topk filter.